marathon/src/main/scala/io/buoyant/marathon/v2/AppIdNamer.scala

```diff
@@ -45,16 +45,23 @@ class AppIdNamer(
       appsActivity.map { apps =>
```

Comment:
To make the types work out, I think you want to change this `map` to `flatMap` and have the function return an `Activity[NameTree[Name]]` (see commend below)